src/audio_resampler.cpp

```diff
@@ -385,7 +385,7 @@ int AudioResampler::FillBuffer(uint8_t* buffer, int length) {
 		int sample_size = AudioDecoder::GetSamplesizeForFormat(output_format);
 
 		// Duplicate data from the back, allows writing to the buffer directly
-		for (size_t i = amount_filled - sample_size; i > 0; i -= sample_size) {
+		for (int i = amount_filled - sample_size; i > 0; i -= sample_size) {
```

Comment:
Why not modifying above (line 380) to `<= 0` ?

Reply:
That's a good question (and this was my first fix):
Because of the memset below:
```cpp

Reply:
...but this is never reached because of the return, no?

Reply:
Wait. Yeah this memset can be removed because the Decode function in AudioDecoder (which calls FillBuffer) handles the memset already... lemme fix this